roomserver/internal/api.go

```diff
@@ -115,6 +108,15 @@ func (r *RoomserverInternalAPI) SetFederationAPI(fsAPI fsAPI.RoomserverFederatio
 		logrus.Panic(err)
 	}
 
+	r.Queryer = &query.Queryer{
+		DB:                r.DB,
+		Cache:             r.Cache,
+		IsLocalServerName: r.Cfg.Global.IsLocalServerName,
+		ServerACLs:        r.ServerACLs,
+		Cfg:               r.Cfg,
+		FSAPI:             fsAPI,
+	}
```

Comment:
I'm surprised this hadn't already been moved here with the rest of the `er`'s. It fits much nicer here :+1:

Reply:
Thanks! I also had to move around some of the initialisation code to make it work (in a later commit, https://github.com/matrix-org/dendrite/pull/3134/commits/67f6e3e0ed63703c0f43c6ad54b0349fbc1b7236 )